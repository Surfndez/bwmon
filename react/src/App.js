import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { DateTime } from 'luxon';
import theme from './theme';
import { ThemeProvider } from '@material-ui/core/styles';
import { CssBaseline } from '@material-ui/core';
import Header from './components/main/Header';
import Footer from './components/main/Footer';
import Navigation from './components/navigation/Navigation';
import Routes from './components/navigation/Routes';
import ErrorBoundary from './components/error/ErrorBoundary';
import menu from './menu/Menu';

import { createBrowserHistory } from 'history';

const basepath = process.env.PUBLIC_URL;

export const history = createBrowserHistory({
    basename: basepath,
});

const App = ({ currentTime }) => (
    <BrowserRouter basename={basepath}>
        <ThemeProvider theme={theme}>
            <CssBaseline />
            <Header />
            <Navigation menu={menu} />
            <ErrorBoundary>
                <Routes menu={menu} />
                <Footer currentTime={currentTime} />
            </ErrorBoundary>
        </ThemeProvider>
    </BrowserRouter>
);

App.defaultProps = {
    currentTime: DateTime.local().toLocaleString(DateTime.DATETIME_FULL_WITH_SECONDS),
};

export default App;
