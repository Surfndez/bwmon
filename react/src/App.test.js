import React from 'react';
import { act, create } from 'react-test-renderer';
import wait from './__test__/utils/Wait';

jest.mock('./services/Usage');

import App from './App';

describe('App', () => {
    const currentTime = 'October 20, 2020, 11:25:35 AM EDT';

    test('renders correctly', () => {
        const tree = create(<App currentTime={currentTime} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    // current issue in Material UI Table Pagination label id
    /*
    test('renders correctly after data load', async () => {
        let tree;
        act(() => {
            tree = create(<App currentTime={currentTime} />);
        });
        await wait(1);
        expect(tree.toJSON()).toMatchSnapshot();
    });
    */
});
