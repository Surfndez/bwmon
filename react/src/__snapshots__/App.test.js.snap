// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
Array [
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      className="MuiToolbar-root MuiToolbar-dense MuiToolbar-gutters"
    >
      <h6
        className="MuiTypography-root makeStyles-title-1 MuiTypography-h6"
        id="app-title"
      >
        Bandwidth Monitor Usage
      </h6>
      <p
        className="MuiTypography-root MuiTypography-body1"
        id="app-version"
      >
        3.0.0
      </p>
    </div>
  </header>,
  <div
    className="MuiBottomNavigation-root"
  >
    <a
      aria-disabled={false}
      className="MuiButtonBase-root MuiBottomNavigationAction-root Mui-selected"
      href="/UsageByUser"
      id="usage-by-user"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        className="MuiBottomNavigationAction-wrapper"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
          />
        </svg>
        <span
          className="MuiBottomNavigationAction-label Mui-selected"
        >
          By User
        </span>
      </span>
    </a>
    <a
      aria-disabled={false}
      className="MuiButtonBase-root MuiBottomNavigationAction-root"
      href="/UsageByUserGraph"
      id="usage-by-user-graph"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        className="MuiBottomNavigationAction-wrapper"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3.5 18.49l6-6.01 4 4L22 6.92l-1.41-1.41-7.09 7.97-4-4L2 16.99z"
          />
        </svg>
        <span
          className="MuiBottomNavigationAction-label"
        >
          By User Graph
        </span>
      </span>
    </a>
    <a
      aria-disabled={false}
      className="MuiButtonBase-root MuiBottomNavigationAction-root"
      href="/UsageByMonth"
      id="usage-by-month"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        className="MuiBottomNavigationAction-wrapper"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 3h-1V1h-2v2H7V1H5v2H4c-1.1 0-2 .9-2 2v16c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 18H4V8h16v13z"
          />
        </svg>
        <span
          className="MuiBottomNavigationAction-label"
        >
          By Month
        </span>
      </span>
    </a>
    <a
      aria-disabled={false}
      className="MuiButtonBase-root MuiBottomNavigationAction-root"
      href="/UsageByMonthGraph"
      id="usage-by-month-graph"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        className="MuiBottomNavigationAction-wrapper"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M23 8c0 1.1-.9 2-2 2-.18 0-.35-.02-.51-.07l-3.56 3.55c.05.16.07.34.07.52 0 1.1-.9 2-2 2s-2-.9-2-2c0-.18.02-.36.07-.52l-2.55-2.55c-.16.05-.34.07-.52.07s-.36-.02-.52-.07l-4.55 4.56c.05.16.07.33.07.51 0 1.1-.9 2-2 2s-2-.9-2-2 .9-2 2-2c.18 0 .35.02.51.07l4.56-4.55C8.02 9.36 8 9.18 8 9c0-1.1.9-2 2-2s2 .9 2 2c0 .18-.02.36-.07.52l2.55 2.55c.16-.05.34-.07.52-.07s.36.02.52.07l3.55-3.56C19.02 8.35 19 8.18 19 8c0-1.1.9-2 2-2s2 .9 2 2z"
          />
        </svg>
        <span
          className="MuiBottomNavigationAction-label"
        >
          By Month Graph
        </span>
      </span>
    </a>
    <a
      aria-disabled={false}
      className="MuiButtonBase-root MuiBottomNavigationAction-root"
      href="/UsageByYear"
      id="usage-by-year"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        className="MuiBottomNavigationAction-wrapper"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 17h18v2H3zm0-7h18v5H3zm0-4h18v2H3z"
          />
        </svg>
        <span
          className="MuiBottomNavigationAction-label"
        >
          By Year
        </span>
      </span>
    </a>
    <a
      aria-disabled={false}
      className="MuiButtonBase-root MuiBottomNavigationAction-root"
      href="/UsageByYearGraph"
      id="usage-by-year-graph"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        className="MuiBottomNavigationAction-wrapper"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M23 8c0 1.1-.9 2-2 2-.18 0-.35-.02-.51-.07l-3.56 3.55c.05.16.07.34.07.52 0 1.1-.9 2-2 2s-2-.9-2-2c0-.18.02-.36.07-.52l-2.55-2.55c-.16.05-.34.07-.52.07s-.36-.02-.52-.07l-4.55 4.56c.05.16.07.33.07.51 0 1.1-.9 2-2 2s-2-.9-2-2 .9-2 2-2c.18 0 .35.02.51.07l4.56-4.55C8.02 9.36 8 9.18 8 9c0-1.1.9-2 2-2s2 .9 2 2c0 .18-.02.36-.07.52l2.55 2.55c.16-.05.34-.07.52-.07s.36.02.52.07l3.55-3.56C19.02 8.35 19 8.18 19 8c0-1.1.9-2 2-2s2 .9 2 2z"
          />
        </svg>
        <span
          className="MuiBottomNavigationAction-label"
        >
          By Year Graph
        </span>
      </span>
    </a>
  </div>,
  <div
    className="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      className="makeStyles-root-3"
    >
      <div
        className="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-indeterminate"
        role="progressbar"
      >
        <div
          className="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Indeterminate"
          style={Object {}}
        />
        <div
          className="MuiLinearProgress-bar MuiLinearProgress-bar2Indeterminate MuiLinearProgress-barColorPrimary"
          style={Object {}}
        />
      </div>
      <div
        className="makeStyles-root-2"
      >
        <div
          className="MuiPaper-root MuiAlert-root MuiAlert-standardInfo MuiPaper-elevation0"
          role="alert"
        >
          <div
            className="MuiAlert-icon"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M11,9H13V7H11M12,20C7.59,20 4,16.41 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20, 12C20,16.41 16.41,20 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10, 10 0 0,0 12,2M11,17H13V11H11V17Z"
              />
            </svg>
          </div>
          <div
            className="MuiAlert-message"
          >
            <div
              className="MuiTypography-root MuiAlertTitle-root MuiTypography-body1 MuiTypography-gutterBottom"
            >
              Loading...
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiTableContainer-root"
    >
      <table
        className="MuiTable-root MuiTable-stickyHeader"
        role={null}
      >
        <thead
          className="MuiTableHead-root"
          role={null}
        >
          <tr
            className="MuiTableRow-root MuiTableRow-head"
            role={null}
          >
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall MuiTableCell-stickyHeader"
              scope="col"
            />
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall MuiTableCell-stickyHeader"
              scope="col"
            />
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall MuiTableCell-stickyHeader"
              colSpan={2}
              scope="col"
            >
              <div
                className="MuiFormControl-root MuiTextField-root"
              >
                <div
                  className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedStart"
                  onClick={[Function]}
                >
                  <div
                    className="MuiInputAdornment-root MuiInputAdornment-positionStart"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                      />
                    </svg>
                  </div>
                  <input
                    aria-invalid={false}
                    autoFocus={false}
                    className="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedStart"
                    disabled={false}
                    id="user-filter"
                    onAnimationStart={[Function]}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    placeholder="IP, MAC or USER"
                    required={false}
                    type="text"
                  />
                </div>
              </div>
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTablePagination-root MuiTableCell-sizeSmall MuiTableCell-stickyHeader"
              colSpan={7}
              minimum={12}
              scope="col"
            >
              <div
                className="MuiToolbar-root MuiToolbar-regular MuiTablePagination-toolbar MuiToolbar-gutters"
              >
                <div
                  className="MuiTablePagination-spacer"
                />
                <p
                  className="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
                >
                  0-0 of 0
                </p>
                <div
                  className="makeStyles-root-4"
                >
                  <button
                    aria-label="first page"
                    className="MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled"
                    disabled={true}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={-1}
                    type="button"
                  >
                    <span
                      className="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden={true}
                        className="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M18.41 16.59L13.82 12l4.59-4.59L17 6l-6 6 6 6zM6 6h2v12H6z"
                        />
                      </svg>
                    </span>
                  </button>
                  <button
                    aria-label="previous page"
                    className="MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled"
                    disabled={true}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={-1}
                    type="button"
                  >
                    <span
                      className="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden={true}
                        className="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M15.41 16.59L10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z"
                        />
                      </svg>
                    </span>
                  </button>
                  <button
                    aria-label="next page"
                    className="MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled"
                    disabled={true}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={-1}
                    type="button"
                  >
                    <span
                      className="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden={true}
                        className="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M8.59 16.59L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
                        />
                      </svg>
                    </span>
                  </button>
                  <button
                    aria-label="last page"
                    className="MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled"
                    disabled={true}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={-1}
                    type="button"
                  >
                    <span
                      className="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden={true}
                        className="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M5.59 7.41L10.18 12l-4.59 4.59L7 18l6-6-6-6zM16 6h2v12h-2z"
                        />
                      </svg>
                    </span>
                  </button>
                </div>
              </div>
            </th>
          </tr>
        </thead>
        <thead
          className="MuiTableHead-root"
          role={null}
        >
          <tr
            className="MuiTableRow-root MuiTableRow-head"
            role={null}
          >
            <th
              aria-sort="ascending"
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeSmall MuiTableCell-stickyHeader"
              id="user-IP"
              scope="col"
            >
              <span
                aria-disabled={false}
                className="MuiButtonBase-root MuiTableSortLabel-root MuiTableSortLabel-active"
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="button"
                tabIndex={0}
              >
                IP
                <span
                  className="makeStyles-visuallyHidden-5"
                >
                  sorted ascending
                </span>
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                  />
                </svg>
              </span>
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeSmall MuiTableCell-stickyHeader"
              id="user-MAC"
              scope="col"
            >
              <span
                aria-disabled={false}
                className="MuiButtonBase-root MuiTableSortLabel-root"
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="button"
                tabIndex={0}
              >
                MAC
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                  />
                </svg>
              </span>
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeSmall MuiTableCell-stickyHeader"
              id="user-user"
              scope="col"
            >
              <span
                aria-disabled={false}
                className="MuiButtonBase-root MuiTableSortLabel-root"
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="button"
                tabIndex={0}
              >
                User
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                  />
                </svg>
              </span>
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeSmall MuiTableCell-stickyHeader"
              scope="col"
            >
              Down
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeSmall MuiTableCell-stickyHeader"
              scope="col"
            >
              Up
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeSmall MuiTableCell-stickyHeader"
              id="user-total"
              scope="col"
            >
              <span
                aria-disabled={false}
                className="MuiButtonBase-root MuiTableSortLabel-root"
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="button"
                tabIndex={0}
              >
                Total
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                  />
                </svg>
              </span>
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeSmall MuiTableCell-stickyHeader"
              scope="col"
            >
              Percent
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeSmall MuiTableCell-stickyHeader"
              scope="col"
            >
              Average
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeSmall MuiTableCell-stickyHeader"
              scope="col"
            >
              Days
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeSmall MuiTableCell-stickyHeader"
              scope="col"
            >
              First Seen
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeSmall MuiTableCell-stickyHeader"
              scope="col"
            >
              Last Seen
            </th>
          </tr>
        </thead>
        <tbody
          className="MuiTableBody-root"
          role={null}
        />
        <thead
          className="MuiTableHead-root"
          role={null}
        >
          <tr
            className="MuiTableRow-root MuiTableRow-head"
            id="user-footer"
            role={null}
          >
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeSmall MuiTableCell-stickyHeader"
              scope="col"
            >
              Totals
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeSmall MuiTableCell-stickyHeader"
              scope="col"
            >
              
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeSmall MuiTableCell-stickyHeader"
              scope="col"
            >
              
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeSmall MuiTableCell-stickyHeader"
              scope="col"
            >
              NaN
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeSmall MuiTableCell-stickyHeader"
              scope="col"
            >
              NaN
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeSmall MuiTableCell-stickyHeader"
              scope="col"
            >
              NaN
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeSmall MuiTableCell-stickyHeader"
              scope="col"
            >
              
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeSmall MuiTableCell-stickyHeader"
              scope="col"
            >
              NaN
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeSmall MuiTableCell-stickyHeader"
              scope="col"
            />
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeSmall MuiTableCell-stickyHeader"
              scope="col"
            >
              
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeSmall MuiTableCell-stickyHeader"
              scope="col"
            >
              
            </th>
          </tr>
        </thead>
      </table>
    </div>
  </div>,
  <h6
    className="MuiTypography-root makeStyles-footer-6 MuiTypography-subtitle2"
  >
    All usage information is in gigabytes
  </h6>,
]
`;
